src/Temporalio/Activities/ActivityServiceProviderAccessor.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Threading;
+
+namespace Temporalio.Activities
+{
+    /// <summary>
+    /// Provides access to the scoped <see cref="IServiceProvider" /> for the
+    /// current activity execution context.
+    /// </summary>
+    public static class ActivityServiceProviderAccessor
+    {
+        /// <summary>
+        /// Gets the async local current value.
+        /// </summary>
+        public static readonly AsyncLocal<IServiceProvider?> AsyncLocalCurrent = new();
```

Comment:
Due to the need to access this across packages, this can't be internal unless `Temporalio` is modified to make internals available to `Temporalio.Extensions.Hosting`. I don't want this to be `public`, but need some guidance on the right way to hide this value from the public.

Reply:
There is also some tension here because an `interface` implementation of `IWorkerInterceptorWithServiceProvider` would actually require that this field (or a wrapping property) is public because applications that are not `NETCOREAPP3_0_OR_GREATER` would need to be able to access this field to replicate this logic
```
ActivityInboundInterceptor InterceptActivity(ActivityInboundInterceptor nextInterceptor) =>
    InterceptActivity(ActivityServiceProviderAccessor.Current, nextInterceptor);
```